### PR TITLE
Add purify radioastronomy benchmarks

### PR DESCRIPTION
**Pull request by tkoskela**
https://github.com/astro-informatics/purify

### CODE REVIEW

**Comment by tkoskela** [comment]
Getting there. As soon as I have released working versions of `sopt` and `purify` this will be ready.

**Comment by tkoskela** [comment]
The purify package PR is still open in spack but otherwise this is ready for review. For testing you for now need to use my branch of spack https://github.com/tkoskela/spack/tree/tk/purify

**Comment by tkoskela** [comment]
purify and dependencies are merged into spack, this should work with develop branch of spack now

File: benchmarks/apps/purify/purify.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+class PurifyFFTBenchmark(PurifyBase):
+    """
+    Execute all the benchmarks in the fft benchmark on one core with one thread
+    """
+
+    num_tasks = 1
+    num_cpus_per_task = 1
+    env_vars = {
+        'OMP_NUM_THREADS': 1
+    }
```
Comment (giordano) [inline benchmarks/apps/purify/purify.py:46]: This overrides `env_vars`, it's better to add items to it instead:
```suggestion
    env_vars['OMP_NUM_THREADS'] = 1
```

File: benchmarks/apps/purify/purify.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    """
+    Execute all the benchmarks in mpi_benchmark_MO on one core with 16 threads
+    (Probably not the greatest idea).
+    """
+
+    num_tasks = 1
+    num_cpus_per_task = 1
+    env_vars = {
+        'OMP_NUM_THREADS': 16
+    }
```
Comment (giordano) [inline benchmarks/apps/purify/purify.py:68]:
```suggestion
    env_vars['OMP_NUM_THREADS'] = '16'
```

**Review by giordano — APPROVED** [review]
Haven't tested myself, but looks good to me!

**Pull request merged**